src/regressionmodel.jl

```diff
@@ -114,6 +114,22 @@ function reconstruct! end
     offset(model::RegressionModel)
 
 Return the offset used in the model, i.e. the term added to the linear predictor with
-known coefficient 1.
+known coefficient 1, or `nothing` if the model was not fit with an offset.
 """
 function offset end
+
+"""
+    linearpredictor(model::RegressionModel)
+
+Return the model's linear predictor, `Xβ` where `X` is the model matrix and `β` is the
+vector of coefficients, or `Xβ + offset` if the model was fit with an offset.
+"""
+function linearpredictor end
+
+"""
+    linearpredictor!([storage,] model::RegressionModel)
+
+In-place version of [`linearpredictor`](@ref), storing the result in `storage` or
+updating `model` directly, if applicable.
```

Comment:
What does updating model directly do? Do you have any examples of this?

Reply:
I was thinking of a case in which the relevant storage lives inside the model object itself. BetaRegression does this, as does GLM, though GLM exposes it in a roundabout way by passing the field as the first argument. I could also imagine a situation where updating a model directly involves mutating multiple internal fields, in which case you couldn't pass only the mutated field as the first argument since there'd be more than one.

Reply:
But when the linear predictor is already stored in the model, `linearpredictor` can just return it directly, there's no need to call that operation `linearpredictor!`, right?

Reply:
OK. I still don't really see what `linearpredictor!(model)` is supposed to do. :-)

I imagine the idea is to follow the pattern used here?
https://github.com/ararslan/BetaRegression.jl/blob/beb83b5ad0c9acbf8cee19138395a49d66be774e/src/BetaRegression.jl#L150
If so, my suggestion is to simply use `linearpredictor(model)` for that, as the fact that internal fields are mutated is an implementation detail (as long as users are clearly told that the returned vector is owned by the model).

Reply:
Yeah, that's the pattern I was referring to, which I was imagining could similarly be adopted in GLM to simplify some calls. For BetaRegression, as long as the linear predictor is stored in the model I want to keep `linearpredictor` purely as an accessor as it's used that way extensively (currently as `linpred`) inside the package. In the meantime I've removed the bit in the documentation here about a single argument.

Reply:
OK, let's go with with this simple version for now then.